### PR TITLE
added doc for numpy operators

### PR DESCRIPTION
**Pull request by endvroy**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: python/mxnet/_numpy_op_doc.py [hunk 1]
```diff
@@ -176,3 +176,229 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+        Input data.
+        Python native iterables not supported.
+
+    axis : None or int or tuple of ints, optional
+        Axis or axes along which to operate. By default, flattened input is
+        used.
+
+        Negative indices not supported.
+
+    out : ndarray, optional
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: Is `out` really the 3rd argument of amax for our version? I believe all MXNet operators put `out` in the end, not in the middle.

File: python/mxnet/_numpy_op_doc.py [hunk 2]
```diff
@@ -176,3 +176,229 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+        passed through to the `amax` method of sub-classes of
+        `ndarray`, however any non-default value will be.  If the
+        sub-class' method does not implement `keepdims` any
+        exceptions will be raised.
+
+    initial : scalar, optional
+        The minimum value of an output element. Must be present to allow
+        computation on empty slice.
+
+        Not supported yet.
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: Actually there's one input supported, you can pass `None` and this operator will still function.

File: python/mxnet/_numpy_op_doc.py [hunk 3]
```diff
@@ -176,3 +176,229 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+           [2, 3]])
+    >>> np.amax(a)           # Maximum of the flattened array
+    array(3)
+    >>> np.amax(a, axis=0)   # Maxima along the first axis
+    array([2, 3])
+    >>> np.amax(a, axis=1)   # Maxima along the second axis
+    array([1, 3])
+    >>> np.amax(a, axis=1, keepdims=True)
+    array([[1],
+          [3]])
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: Please make sure the alignment of the print-outs look correct on the rendered webpage.

File: python/mxnet/_numpy_op_doc.py [hunk 4]
```diff
@@ -176,3 +176,229 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+    <https://docs.scipy.org/doc/numpy/reference/generated/numpy.squeeze.html>`_ in
+    the following aspects:
+
+    - The default value type is `float32` instead of `float64` in numpy.
+    - `a` only supports ndarray.
+    """
+    pass
+
+
+def _npi_ones(shape, dtype=None, order='C'):
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: Is this an npi operator with a wrapper? If yes then the wrappers' documents will be enough and you do not need to put extra doc in this file.

File: python/mxnet/_numpy_op_doc.py [hunk 5]
```diff
@@ -176,3 +176,229 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
...
+    This function differs to the original `numpy.ones
+    <https://docs.scipy.org/doc/numpy/reference/generated/numpy.ones.html>`_ in
+    the following aspects:
+
+    - The default value type is `float32` instead of `float64` in numpy.
+    - `order` is not supported.
+    """
+
+
+def _npi_random_uniform(low=0.0, high=1.0, size=None):
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: This seems to be the same case as `_npi_ones`.

File: python/mxnet/ndarray/numpy/_op.py [hunk 6]
```diff
@@ -882,3 +883,172 @@ def sqrt(x, out=None, **kwargs):
...
+    words ``floor(-2.5) == -2``.  MXNet instead uses the definition of
+    `floor` where `floor(-2.5) == -3`.
+
+    This function differs to the original `numpy.floor
+    <https://docs.scipy.org/doc/numpy/reference/generated/numpy.floor.html>`_ in
+    the following aspects:
+
+    - The default value type is `float32` instead of `float64` in numpy.
+    - `a` only supports ndarray.
+    - `a` doe snot support scalar.
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: "does not". and have you tried this operator with a scalar input? I believe the current `_unary_func_helper` was written to enable such support...

File: python/mxnet/ndarray/numpy/_op.py [hunk 7]
```diff
@@ -882,3 +883,172 @@ def sqrt(x, out=None, **kwargs):
...
+        Input array.
+
+    out : ndarray, None, or tuple of exactly one ndarray, optional
+
+        A location into which the result is stored. If provided, it must have
+        a shape that the inputs broadcast to. If not provided or `None`,
+        a freshly-allocated array is returned. A tuple (possible only as a
+        keyword argument) must have exactly one element.
+
+    where : array_like, optional
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: you can override the signature, and do not expose an argument if it's totally not supported. Same comment for all other similar ops.

File: python/mxnet/_numpy_op_doc.py (hunk 1 above)
Comment (endvroy) [reply python/mxnet/_numpy_op_doc.py]: fixed

File: python/mxnet/_numpy_op_doc.py (hunk 3 above)
Comment (endvroy) [reply python/mxnet/_numpy_op_doc.py]: fixed

File: python/mxnet/_numpy_op_doc.py (hunk 4 above)
Comment (endvroy) [reply python/mxnet/_numpy_op_doc.py]: fixed

File: python/mxnet/_numpy_op_doc.py (hunk 2 above)
Comment (endvroy) [reply python/mxnet/_numpy_op_doc.py]: fixed

File: python/mxnet/ndarray/numpy/_op.py (hunk 6 above)
Comment (endvroy) [reply python/mxnet/ndarray/numpy/_op.py]: It supports Python int but not scalar ndarray. Docs are updated to reflect the status.

File: python/mxnet/ndarray/numpy/_op.py (hunk 7 above)
Comment (endvroy) [reply python/mxnet/ndarray/numpy/_op.py]: removed redundant parameter.

File: python/mxnet/_numpy_op_doc.py (hunk 5 above)
Comment (endvroy) [reply python/mxnet/_numpy_op_doc.py]: fixed

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-response]

File: python/mxnet/ndarray/numpy/_op.py [hunk 8]
```diff
@@ -882,3 +906,162 @@ def sqrt(x, out=None, **kwargs):
...
+@set_module('mxnet.ndarray.numpy')
+def floor(x, out=None, **kwargs):
+    r"""
+    Return the floor of the input, element-wise.
+    The floor of the scalar `x` is the largest integer `i`, such that
+    `i <= x`.  It is often denoted as :math:`\lfloor x \rfloor`.
+
+    Parameters
+    ----------
+    x : ndarray or Python int
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: Are you sure this accepts 'Python int' only except ndarrays? Does `floor(2.5)` not work?

File: python/mxnet/ndarray/numpy/_op.py [hunk 9]
```diff
@@ -882,3 +906,162 @@ def sqrt(x, out=None, **kwargs):
...
+
+        A location into which the result is stored. If provided, it must have
+        a shape that the inputs broadcast to. If not provided or `None`,
+        a freshly-allocated array is returned. A tuple (possible only as a
+        keyword argument) must have exactly one element. If `x` is a Python int,
+        nothing will be written to `out`.
+
+    Returns
+    -------
+    out : ndarray or int
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: same here, have you made sure that python native floating point numbers does not work for this op?

File: python/mxnet/symbol/numpy/_symbol.py [hunk 10]
```diff
@@ -1555,4 +1556,112 @@ def sqrt(x, out=None, **kwargs):
...
+    The floor of the scalar `x` is the largest integer `i`, such that
+    `i <= x`.  It is often denoted as :math:`\lfloor x \rfloor`.
+
+    Parameters
+    ----------
+    x : _Symbol
+        Input data.
+        Does not support scalar.
+    out : _Symbol or None, optional
+    where : array_like, optional
```
Comment (haojin2) [inline python/mxnet/symbol/numpy/_symbol.py]: If it's totally not supported then do not put it in the `Parameters` section. Note this down in the notes section.

File: python/mxnet/symbol/numpy/random.py [hunk 11]
```diff
@@ -86,6 +86,7 @@ def uniform(low=0.0, high=1.0, size=None, **kwargs):
     Notes
     -----
     This function currently does not support ``low`` and ``high`` as symbols.
+    `size` is not optional.
```
Comment (haojin2) [inline python/mxnet/symbol/numpy/random.py]: I'd rather remove the `, optional` part in the `Parameter` section above for `size`.

Comment (endvroy) [reply python/mxnet/symbol/numpy/random.py]: fixed and updated notes

File: python/mxnet/symbol/numpy/_symbol.py (hunk 10 above)
Comment (endvroy) [reply python/mxnet/symbol/numpy/_symbol.py]: fixed and moved to notes.

File: python/mxnet/ndarray/numpy/_op.py (hunk 9 above)
Comment (endvroy) [reply python/mxnet/ndarray/numpy/_op.py]: fixed

File: python/mxnet/ndarray/numpy/_op.py (hunk 8 above)
Comment (endvroy) [reply python/mxnet/ndarray/numpy/_op.py]: fixed

File: python/mxnet/ndarray/numpy/_op.py [hunk 12]
```diff
@@ -84,6 +85,29 @@ def ones(shape, dtype=None, **kwargs):
     -------
     out : ndarray
         Array of zeros with the given shape, dtype, and ctx.
+
```
Comment (mikemwx) [inline python/mxnet/ndarray/numpy/_op.py]: The signature of ones should be overridden to expose the the parameter `ctx`

**Pull request closed without merging**